Tests/ApolloCodegenTests/FileManagerExtensionsTests.swift

```diff
@@ -366,78 +390,129 @@ class FileManagerExtensionTests: XCTestCase {
+      expect(isDirectory).notTo(beNil())
+
+      isDirectory?.pointee = true
+      return false // exists
+
+    }, createFile: { (path: String, data: Data?, attr: FileAttributes?) in
       expect(path).to(match(self.uniquePath))
       expect(data).to(equal(self.uniqueData))
       expect(attr).to(beNil())
 
       return true
-    }, createDirectory: { (path: String, createIntermediates: Bool, attr: FileAttributes?) in
-      let parentPath = URL(fileURLWithPath: path).deletingLastPathComponent().path
 
+    }, createDirectory: { (path: String, createIntermediates: Bool, attr: FileAttributes?) in
       expect(path).to(match(parentPath))
       expect(createIntermediates).to(beTrue())
       expect(attr).to(beNil())
     })
 
     // then
     expect(try mocked.apollo.createFile(atPath: self.uniquePath, data:self.uniqueData)).to(beTrue())
-    expect(mocked.blocksCalled).to(equal([.createDirectory, .createFile]))
+    expect(mocked.blocksCalled).to(equal([.fileExists, .createDirectory, .createFile]))
   }
 
-  func test_createFile_givenTrueFromCreateFileAndThrowFromCreateDirectory_shouldCallCreateDirectoryAndThrow() throws {
+  func test_createFile_givenContainingDirectoryDoesNotExistAndFileNotCreated_shouldReturnFalse() throws {
     // given
-    let mocked = MockFileManager(createFile: { (path: String, data: Data?, attr: FileAttributes?) in
+    let parentPath = URL(fileURLWithPath: self.uniquePath).deletingLastPathComponent().path
+    let mocked = MockFileManager(fileExists: { (path: String, isDirectory: UnsafeMutablePointer<ObjCBool>?) in
+      expect(path).to(match(parentPath))
+      expect(isDirectory).notTo(beNil())
+
+      isDirectory?.pointee = true
+      return false // exists
+
+    }, createFile: { (path: String, data: Data?, attr: FileAttributes?) in
       expect(path).to(match(self.uniquePath))
       expect(data).to(equal(self.uniqueData))
       expect(attr).to(beNil())
 
-      return true
+      return false
+
     }, createDirectory: { (path: String, createIntermediates: Bool, attr: FileAttributes?) in
-      let parentPath = URL(fileURLWithPath: path).deletingLastPathComponent().path
+      expect(path).to(match(parentPath))
+      expect(createIntermediates).to(beTrue())
+      expect(attr).to(beNil())
+    })
 
+    // then
+    expect(try mocked.apollo.createFile(atPath: self.uniquePath, data:self.uniqueData)).to(beFalse())
+    expect(mocked.blocksCalled).to(equal([.fileExists, .createDirectory, .createFile]))
+  }
+
+  func test_createFile_givenContainingDirectoryDoesNotExistAndError_shouldThrow() throws {
+    // given
+    let parentPath = URL(fileURLWithPath: self.uniquePath).deletingLastPathComponent().path
+    let mocked = MockFileManager(fileExists: { (path: String, isDirectory: UnsafeMutablePointer<ObjCBool>?) in
+      expect(path).to(match(parentPath))
+      expect(isDirectory).notTo(beNil())
+
+      isDirectory?.pointee = true
+      return false // exists
```

Comment:
why does `false` mean that it exists? Or am I misunderstanding the point of this comment? same question throughout this file

Reply:
Hmm, no it's not meant to convey that meaning. This is a remnant of the previously named test methods which since they're better named now I can remove these `// exists` comments.

Reply:
Done in 58a98cfa.